subsys/fs/fat_fs.c

```diff
@@ -270,10 +270,12 @@ static int fatfs_readdir(struct fs_dir_t *zdp, struct fs_dirent *entry)
 
 	res = f_readdir(zdp->dirp, &fno);
 	if (res == FR_OK) {
-		entry->type = ((fno.fattrib & AM_DIR) ?
-			       FS_DIR_ENTRY_DIR : FS_DIR_ENTRY_FILE);
 		strcpy(entry->name, fno.fname);
```

Comment:
Does worth to use strncpy with FATFS_MAX_FILE_NAME?

Reply:
The destination buffer is allocated to `MAX_FILE_NAME + 1` where `MAX_FILE_NAME` is either 12 or 256 depending on what filesystems are enabled.

`FATFS_MAX_FILE_NAME` is always 12.

The `FILINFO` struct source buffer will be either 13 or (if LFN is enabled) `1+_MAX_LFN` and it will always be NUL-terminated.

LFN is not enabled in the standard Zephyr FatFS configuration.

So, unless LFN is enabled (in which case this wrapper needs to be updated) IMO this is fine as it was originally.  If we switch to `strncpy` just for safety then we also have to handle the case where the copied value isn't NUL-terminated.

I'd rather leave it as is.

OK?

Reply:
ok